plugins/bcm57xx/fu-bcm57xx-firmware.c

```diff
@@ -0,0 +1,571 @@
+/*
+ * Copyright (C) 2018-2020 Evan Lojewski
+ * Copyright (C) 2020 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include "fu-common.h"
+
+#include "fu-bcm57xx-common.h"
+#include "fu-bcm57xx-firmware.h"
+#include "fu-bcm57xx-dict-image.h"
+#include "fu-bcm57xx-stage1-image.h"
+#include "fu-bcm57xx-stage2-image.h"
+
+#include "fwupd-error.h"
+
+struct _FuBcm57xxFirmware {
+	FuFirmware		 parent_instance;
+	guint16			 vendor;
+	guint16			 model;
+	guint32			 phys_addr;
+	gsize			 source_size;
+	guint8			 source_padchar;
+};
+
+G_DEFINE_TYPE (FuBcm57xxFirmware, fu_bcm57xx_firmware, FU_TYPE_FIRMWARE)
+
+#define BCM_STAGE1_HEADER_MAGIC_BROADCOM	0x0E000E03
+#define BCM_STAGE1_HEADER_MAGIC_MEKLORT		0x3C1D0800
+
+#define BCM_APE_HEADER_MAGIC			0x42434D1A
```

Comment:
With the APE format now matching the NVM format, I'm seeing 0x1A4D4342 here.

Reply:
Ack, thanks.